### ISSUE TITLE
Network module

### ISSUE THREAD

**Issue by hackedWifi** [polybar/polybar#1822]
## Describe the issue
*A clear and concise description of what the bug is.*
Network module doesn't update ESSID when connected
**Expected behavior:**
Update the ESSID of the current connection.
*A clear and concise description of what you expected to happen.*
Show current network ESSID when connected
**Actual behavior:**
It displays previous ESSID used. If I restart the bar it will update to the current connection. Otherwise, will keep displaying previously used connection
*What actually happens*

**Was it working before?**
* Did you also experience this bug in an earlier version of polybar?
* If no, what was the last version where this worked correctly?

## To Reproduce
*A minimal but complete config with which the problem occurs:*
```dosini

```
*List any other steps needed to reproduce the issue besides starting polybar with the config you posted above.*

## Polybar Log
*Post everything polybar outputs to the terminal when you run it and the issue occurs below*
```
No errors when starting polybar
```

## Screenshots
*If applicable, add screenshots to help explain your problem.*

## Environment:
* WM: bspwm
* Distro: Archlinux
* Output of `polybar -vvv`:
polybar 3.3.0-92-g7724c4dc

Features: +alsa +curl -i3 -mpd +network(libnl) +pulseaudio +xkeyboard

X extensions: +randr (+monitors) -render -damage -sync +composite +xkb +xrm +xcursor

Build type: Release
Compiler: /usr/bin/c++
Compiler flags: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -Wall -Wextra -Wpedantic -O3 -DNDEBUG
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now 
## Additional context
*Add any other context that you think is necessary about the problem here.*

**Comment by patrick96** [comment]
So when you connect to another network it doesn't update the essid?

Could you please add steps to reproduce this if you are doing anything out of the ordinary. For me it always updated the ssid when the network changed.

**Comment by hackedWifi** [comment]
I am not doing anything out of the ordinary. If I connect to a wireless connection. It always shows the previous used connection. For example I go to a coffee shop in the morning, then I go back home and connect there. It shows the ESSId from the coffee shop. Unless I reload polybar

**Comment by patrick96** [comment]
This does look like a bug. This has never happened to me before, at least not as far as I'm aware.

Could you check that the ssid is properly updated in the output of `iw dev wlp3s0 info` where you replace `wlp3s0` with your own wireless interface.

Are other things like upspeed and downspeed also not updating in the network module?

**Comment by hackedWifi** [comment]
Yes, it is properly updated. Yet my bar is showing the one I used a day ago. LitteBigWifi5 is the one previously used.

`iw dev wlp2s0 info`
Interface wlp2s0
        ifindex 7
        wdev 0x500000001
        addr 44:1c:a8:e1:db:13
        ssid hackedwifi
        type managed
        wiphy 5
        channel 161 (5805 MHz), width: 20 MHz, center1: 5805 MHz
        txpower 31.00 dBm

**Comment by patrick96** [comment]
> Are other things like upspeed and downspeed also not updating in the network module?

Could you answer this question. It will help us track down what exactly is going wrong.

**Comment by hackedWifi** [comment]
Yes, those work. The SSID doesn't update as reflected on the link below. I am connected to PEETS and polybar shows the previous SSID used about a day ago.
https://imgur.com/a/Axvl3PG

**Comment by evan-goode** [comment]
I'm also experiencing this bug. If it helps, I've noticed it on my MacbookPro15,5 running Arch Linux but not my T480s running Fedora. Both machines are have polybar 3.4.0.

I can reliably reproduce the issue on Arch with the following steps:
1. Have two wireless networks saved in NetworkManager, network A and network B.
2. Connect to network A
3. `systemctl suspend`
4. Either move out of range of network A or stop it from broadcasting (just tested with a mobile hotspot)
5. Make sure you're in range of network B
6. Resume the system. The machine will connect to network B since it can no longer find network A, but polybar will still display network A's SSID.

At this point, I can disconnect from and reconnect to network B, switch back to network A, switch to a third network C, or even restart NetworkManager, but polybar will still be stuck on network A until I kill it.

**Comment by hackedWifi** [comment]
> At this point, I can disconnect from and reconnect to network B, switch back to network A, switch to a third network C, or even restart NetworkManager, but polybar will still be stuck on network A until I kill it.
I still have this issue. I have not been able to find a way to resolve it.

**Comment by Lomadriel** [comment]
I tried to reproduce the issue but I was unable to do so.

However could I ask someone who has that bug to launch polybar with the following command and try to reproduce this bug. 
```zsh
NLCB=debug NLDBG=4 polybar -l trace [rest of your command]
```

Please send me the full log (you can replace your IP, SSID by fake info if you want). Just avoid to launch polybar for a long time as it'll produce a lot of logs.
I'll try to figure what I can do with that.

**Comment by evan-goode** [comment]
Hope this can help: https://gist.github.com/evan-goode/3c0b957b1d664b01e49fc9738d3bf85a

While the log was running, I followed steps 1 through 6 as described above, then ran `systemctl restart NetworkManager`.
